server/api/attendenceApi.js

```diff
@@ -0,0 +1,158 @@
+const request = require('superagent');
+const moment = require('moment');
+const { LEAVE_STATE_CODES } = require('../../src/data/LeaveState');
+const { findUserLeavesForRangeByEmail } = require('../service/leaveService');
+var auth = require('basic-auth');
+const TOGGL_REPORTS_DETAILS_URL = 'https://toggl.com/reports/api/v2/details';
+
+const httpBasicAuthentication = async (req, res, next) => {
+  const setUnauthorized = () => res.status(401).end();
+  var credentials = auth(req);
+  if (credentials) {
+    const isAuthSet = process.env.M2M_USERNAME && process.env.M2M_PASSWORD;
+    const isUsernameRight = credentials.name === process.env.M2M_USERNAME;
+    const isPasswordRight = credentials.pass === process.env.M2M_PASSWORD;
+    if (isAuthSet && isUsernameRight && isPasswordRight) {
+      next();
+    } else {
+      setUnauthorized();
+    }
+  } else {
+    setUnauthorized();
+  }
+};
+
+const setupAttendenceApi = app => {
+  app.use('/api/*', httpBasicAuthentication);
+  app.get('/api/attendence/:month', getAttendence);
+};
+
+const getUserDataFromSSO = async () => {
+  try {
+    const ssoData = await request.get(process.env.SSO_URI).auth(process.env.SSO_USERNAME, process.env.SSO_PASSWORD);
+    return ssoData.body.map(user => {
+      const u = {};
+      u.name = user.name;
+      u.email = user.email;
+      user.keys.forEach(kv => u[kv.key] = kv.value);
+      return u;
+    });
+  } catch (err) {
+    console.log(err);
+    return [];
+  }
+};
+
+const initAttendenceData = (end, ssoUserData) => {
+  const attendence = {};
+
+  const userIdToNameMap = {};
+  ssoUserData.filter(user => user.togglUserId).forEach(user => {
+    const days = [];
+    for (let i = 0; i < end.date(); i++) {
+      days.push({
+        logged_time: 0,
+        leave: '-'
+      });
+    }
+    attendence[user.togglUserId] = {
+      days,
+      total: 0,
+      contractType: user.contractType
+    };
+    userIdToNameMap[user.togglUserId] = user.name;
+  });
+
+  return { attendence, userIdToNameMap };
+};
+
+const writeLeaveToAttendence = (attendence, month, userId, leave) => {
+  let leaveTypeString;
+  switch(leave.type) {
+  case 'ANNUAL': 
+    leaveTypeString = 'D';
+    break;
+  case 'SICKNESS': 
+    leaveTypeString = 'PN';
+    break;
+  case 'DOCTOR': 
+    leaveTypeString = 'L';
+    break;
+  default: leaveTypeString = 'I';
+  }
+
+  const m = moment(leave.startDate);
+  while(m.toDate() < leave.endDate) {
+    if (m.month() === month) attendence[userId].days[m.date() - 1].leave = leaveTypeString;
```

Comment:
I am not fan of not using {} with if. + pls handle multiple absence types in one day (maybe concatenated by space or `+`)